pyccel/ast/numpyext.py

```diff
@@ -2087,6 +2090,83 @@ def __new__(cls, a, axis = None):
 
         return PyccelArrayShapeElement(a, axis)
 
+class NumpyIsNan(PyccelInternalFunction):
+    """ 
+    Represents a call to numpy.isnan() function.
+
+    This class encapsulates a call to the Numpy 'isnan' function. It is used to
+    check whether the elements of a given array or expression are NaN (Not-a-Number).
+
+    Parameters
+    ----------
+    arg : TypedAstNode
+        A Pyccel expression or array to be checked for NaN values.
+
+    See Also
+    --------
+    numpy.isnan :
+        See NumPy docs : <https://numpy.org/doc/stable/reference/generated/numpy.isnan.html>.
+    """
+    __slots__ = ('_arg',)
+    name = 'isnan'
+    _dtype = NativeFloat()
```

Comment:
These classes need to define all the properties described in the class `TypedAstNode` from which they inherit via `PyccelInternalFunction` (you may need to update your fork to get the right names).
The properties are:
- datatype
- precision
- rank
- order

They describe the type of the result of the function call. So I think here you should have `_dtype = NativeBool()` and `_precision = -1.
Anything that is always the same (e.g. dtype and precision) can be defined in the class variables as you have done here. Anything else (probably rank and order) will need to be deduced from the arguments (which should already have these properties) and stored in the class instance.

I'll try and find some time to write you a doc about this ASAP

Reply:
See e.g.
https://github.com/pyccel/pyccel/blob/267be30007685fb23fa799be48c782fb144e46eb/pyccel/ast/numpyext.py#L1489-L1532

Reply:
You may even be able to inherit from this class and just override the dtype setter